Topics/Tech_Stacks/Cypress.md

```diff
@@ -0,0 +1,65 @@
+## E2E Testing with Cypress
+
+# Cypress Introduction
+
+Cypress is mainly used for testing web applications, especially those built in javascript. It provides an interface to programatically test your application, and visually what went wrong (or right) in tests. 
+  
+# Why do end to end testing?  
+  
+[https://circleci.com/blog/what-is-end-to-end-testing/](https://circleci.com/blog/what-is-end-to-end-testing/)
+
+The above link has a good explanation on what end to end testing is and why it should be used. 
+
+Cypress very closely mimics a real user, think of it as a robot accessing your website from a browser like a human would, but you can program the robot to interact with your website however you like and programmatically check the output on the screen.
```

Comment:
It might be beneficial to add a comparison with other testing tools after this section, to highlight the strengths and weaknesses of Cypress. Or, even after the "Best Practices" section.

Reply:
Also a great idea

Reply:
Let me know if this section is hard to understand.

Reply:
I think it's easy enough to understand. If you want, an example might also help explain.